### PR TITLE
Add test for canceling one branch when stream closes or errors

### PR DESCRIPTION
**Pull request by MattiasBuelens**
See whatwg/streams#1039 for the accompanying spec change.

### CODE REVIEW

File: streams/readable-streams/tee.any.js [hunk 1]
```diff
@@ -317,6 +317,60 @@ promise_test(t => {
 
 }, 'ReadableStream teeing: erroring the original should immediately error the branches');
 
+promise_test(t => {
```
Comment (youennf) [inline streams/readable-streams/tee.any.js:320]: Could use async/await to ease readability

File: streams/readable-streams/tee.any.js [hunk 2]
```diff
@@ -317,6 +317,60 @@ promise_test(t => {
...
+      controller = c;
+    }
+  });
+
+  const [reader1, reader2] = rs.tee().map(branch => branch.getReader());
+  const cancelPromise = reader2.cancel();
+
+  controller.enqueue('a');
+
+  return flushAsyncEvents().then(() => {
```
Comment (youennf) [inline streams/readable-streams/tee.any.js:334]: Why is it needed here and below?

**Review by youennf — APPROVED** [review]

File: streams/readable-streams/tee.any.js (hunk 2 above)
Comment (domenic) [reply streams/readable-streams/tee.any.js:334]: +1, `flushAsyncEvents()` should always come with a comment at least.

File: streams/readable-streams/tee.any.js (hunk 1 above)
Comment (domenic) [reply streams/readable-streams/tee.any.js:320]: The original streams tests were written without `async`/`await` but I support using it going forward.

Comment (MattiasBuelens) [reply streams/readable-streams/tee.any.js:320]: I went with promise chaining for consistency with the other tests in the same file. But if you're okay with writing new tests with `async`/`await`, then I'll quickly rework them. (Maybe we can refactor the existing tests in a separate PR? 😁)

File: streams/readable-streams/tee.any.js (hunk 2 above)
Comment (MattiasBuelens) [reply streams/readable-streams/tee.any.js]: You're right, it's not needed. I'll remove it.

**Pull request merged**